amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -99,67 +104,48 @@ public TaskResult call() {
+                        final ExtendedSequenceNumber lastCheckpointValue = recordProcessorCheckpointer.lastCheckpointValue();
                         if (lastCheckpointValue == null
                                 || !lastCheckpointValue.equals(ExtendedSequenceNumber.SHARD_END)) {
                             throw new IllegalArgumentException("Application didn't checkpoint at end of shard "
-                                    + shardInfoIdProvider.apply(shardInfo) + ". Application must checkpoint upon shard end. " +
-                                    "See ShardRecordProcessor.shardEnded javadocs for more information.");
+                                                                       + shardInfoIdProvider.apply(shardInfo) + ". Application must checkpoint upon shard end. " +
+                                                                       "See ShardRecordProcessor.shardEnded javadocs for more information.");
                         }
-                    } else {
+                    } catch (Exception e) {
+                        applicationException = true;
+                        throw e;
+                    } finally {
+                        MetricsUtil.addLatency(scope, RECORD_PROCESSOR_SHUTDOWN_METRIC, startTime, MetricsLevel.SUMMARY);
+                    }
+                } else {
+                    try {
                         shardRecordProcessor.leaseLost(LeaseLostInput.builder().build());
+                    } catch (Exception e) {
+                        applicationException = true;
+                        throw e;
                     }
-                    log.debug("Shutting down retrieval strategy.");
-                    recordsPublisher.shutdown();
-                    log.debug("Record processor completed shutdown() for shard {}", shardInfoIdProvider.apply(shardInfo));
-                } catch (Exception e) {
-                    applicationException = true;
-                    throw e;
-                } finally {
-                    MetricsUtil.addLatency(scope, RECORD_PROCESSOR_SHUTDOWN_METRIC, startTime, MetricsLevel.SUMMARY);
                 }
 
-                if (localReason == ShutdownReason.SHARD_END) {
-                    log.debug("Looking for child shards of shard {}", shardInfoIdProvider.apply(shardInfo));
-                    // create leases for the child shards
-                    hierarchicalShardSyncer.checkAndCreateLeaseForNewShards(shardDetector, leaseCoordinator.leaseRefresher(),
-                            initialPositionInStream, latestShards, cleanupLeasesOfCompletedShards, ignoreUnexpectedChildShards, scope, garbageCollectLeases,
-                            isLeaseTableEmpty);
-                    log.debug("Finished checking for child shards of shard {}", shardInfoIdProvider.apply(shardInfo));
-                }
+                log.debug("Shutting down retrieval strategy.");
```

Comment:
Need to log the state : shardInfoIdProvider.apply(shardInfo)

Reply:
Addressed